Documentation/nfs.md

```diff
@@ -0,0 +1,222 @@
+```
+
+## Create and Initialize NFS Server
+
+Now that the operator is running, we can create an instance of a NFS server by creating an instance of the `nfsservers.nfs.rook.io` resource.
+The resource values are used to configure the NFS server and export.
+Full details of the configuration option can be found in the [NFS CRD documentation](nfs-crd.md).
+
+## Hostpath example
+Check the content of the CRD instance yaml file
+```console
+$ cat nfs-hostpath.yaml
+
+apiVersion: v1
+kind: Namespace
+metadata:
+  name:  rook-nfs
+---
+# A default storageclass must be present
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: nfs-hostpath-claim
+  namespace: rook-nfs
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: nfs.rook.io/v1alpha1
+kind: NFSServer
+metadata:
+  name: rook-nfs
+  namespace: rook-nfs
+spec:
+  replicas: 1
+  exports:
+  - name: nfs-share
+    server:
+      accessMode: ReadWrite
+      squash: "none"
+    # A Persistent Volume Claim must be created before creating NFS CRD instance.
+    persistentVolumeClaim:
+      claimName: nfs-hostpath-claim
+```
+
+
+When you are ready to create a nfs export, simply run:
+
+```console
+$ kubectl create -f nfs-hostpath.yaml
+```
+
+We can verify that a Kubernetes object has been created that represents our new NFS export with the command below.
+
+```console
+$ kubectl -n rook-nfs get nfsservers.nfs.rook.io
+
+NAME       AGE
+rook-nfs   1m
+```
+
+To check if the NFS server is running:
+
+```console
+$ kubectl -n rook-nfs get pod -l app=rook-nfs
+
+NAME         READY     STATUS    RESTARTS   AGE
+rook-nfs-0   1/1       Running   0          2m
+```
+
+## Accessing the Export
+
+To access the export, first create a persistent volume:
+To create the persistent volume, it is required to get the valid cluster ip of the nfs server pod.
+
+Then edit the ip in nfs-pv.yaml
+
+```console
+$ cat pv.yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: rook-nfs-pv
+  namespace: rook-nfs
+spec:
+  capacity:
+    storage: 1Mi
+  accessModes:
+    - ReadWriteMany
+  nfs:
+    # FIXME: use the right IP
+    # run the bellow command to get the valid IP
+    # $ kubectl -n rook-nfs get service -l app=rook-nfs
+    server: <Cluster IP>
+    path: "/<Claim name>"
+
+```
+Get the valid ip of the nfs server using the following command and replace the `<Cluster IP>` as in the above definition with that IP
+```console
+$ kubectl -n rook-nfs get service -l app=rook-nfs
+```
+Replace `<Claim name>` with claimName as configured in the nfs-hostpath.yaml. So in this case it will be `nfs-hostpath-claim` and it will look like `path: "/nfs-hostpath-claim"`
+Now create the PV
+
+```console
+$ kubectl create -f pv.yaml
+```
+Then a PVC is used to access this storage.
+
+## Consuming the Export
+
+There is an example of a web server app that utilizes the NFS Export to store data of its document root.
```

Comment:
We don't have the `pvc.yaml` content in this document, that needs to be included too.

wording/grammar:

Now we can consume the PV that we just created by creating an example web server app that uses a PVC to claim the exported volume.  First, create the PVC object: